cli/src/main/kotlin/ls/backup/cli/Backup.kt

```diff
@@ -0,0 +1,217 @@
+package ls.backup.cli
+
+import aws.sdk.kotlin.runtime.auth.credentials.ProfileCredentialsProvider
+import aws.sdk.kotlin.services.s3.S3Client
+import aws.sdk.kotlin.services.s3.headBucket
+import aws.sdk.kotlin.services.s3.listObjectsV2
+import aws.sdk.kotlin.services.s3.model.GetObjectRequest
+import aws.smithy.kotlin.runtime.content.writeToFile
+import aws.smithy.kotlin.runtime.net.Url
+import kotlinx.coroutines.runBlocking
+import ls.kafka.model.DumpRecord
+import ls.kafka.model.Record
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.common.serialization.ByteArraySerializer
+import java.io.DataInputStream
+import java.io.EOFException
+import java.io.InputStream
+import java.util.*
+import kotlin.io.path.createTempFile
+import kotlin.io.path.inputStream
+
+data class BackupFilePath(val name: String) {
+    val topic: String
+    val partition: Int
+    val startOffset: Long
+
+    init {
+        name.substringBefore(".").split("+").let { nameParts ->
+            topic = nameParts[0]
+            partition = nameParts[1].toInt()
+            startOffset = nameParts[2].toLong()
+        }
+    }
+
+    override fun toString(): String {
+        return "BackupFilePath(name='$name')"
+    }
+}
+
+class BackupFile(name: String, inputStream: InputStream) {
+
+    val compressed = name.endsWith(".gz")
+
+    val topic: String
+    val partition: Int
+    val startOffset: Long
+
+    private val stream: InputStream
+
+    init {
+        name.split("/").last().substringBefore(".").split("+").let { nameParts ->
+            topic = nameParts[0]
+            partition = nameParts[1].toInt()
+            startOffset = nameParts[2].toLong()
+        }
+        stream = if (compressed) {
+            GzipCompressorInputStream(inputStream)
+        } else {
+            inputStream
+        }
+    }
+
+    fun records(): Sequence<DumpRecord> {
+        return stream.loadDump()
+    }
+}
+
+
+fun DumpRecord.withTopic(topic: String) = Record(topic, partition, offset, ts, key, value)
+
+fun InputStream.loadDump(): Sequence<DumpRecord> {
```

Comment:
Please move the stream reader logic to the `:format` subproject. I could imagine to define next to the `ls.kafka.io.RecordStreamWriter`  a new `io.kafka.io.RecordStreamReader` that knows how to read a record.

Ensure that the new Reader implementation is generic without any connector dependency.